src/Octoshift/Handlers/DownloadLogsCommandHandler.cs

```diff
@@ -88,11 +78,33 @@ public async Task Handle(DownloadLogsCommandArgs args)
             throw new OctoshiftCliException($"Migration log for repository {args.GithubRepo} unavailable!");
         }
 
-        var logUrl = result.Result;
+        var (logUrl, migrationId) = result.Result.Value;
+
+        args.MigrationLogFile ??= $"migration-log-{args.GithubOrg}-{args.GithubRepo}-{migrationId}.log";
+
+        // We already checked if the file exists above for the case where the user explicitly picked their own
+        // filename. This handles the case where the filename has been set to the default based on the inputs
+        // and migration ID.
+        CheckIfOutputFileAlreadyExists(args.MigrationLogFile, args.Overwrite);
 
         _log.LogInformation($"Downloading log for repository {args.GithubRepo} to {args.MigrationLogFile}...");
         await _httpDownloadService.DownloadToFile(logUrl, args.MigrationLogFile);
 
         _log.LogSuccess($"Downloaded {args.GithubRepo} log to {args.MigrationLogFile}.");
     }
+
+    private void CheckIfOutputFileAlreadyExists(string outputPath, bool shouldOverwrite)
```

Comment:
I wouldn't create this method (I've already added code snippets w/o using it). Also the if logic is wrong here an in case of an overwrite this is going to throw! the fact that no unit test is broken by changing the logic is a sign that we either don't have enough tests some tests are wrong. We def. need to add/fix tests for this. 
If you want to be DRY not to repeat the exception message you can create a method to throw the exception or maybe create one for building the error message, something like this:
```c#
throw new OctoshiftCliException(CreateFileAlreadyExistingErrorMessage(outputPath));
```

Reply:
This is the opposite of the intended logic 😂 Definitely needs better tests - need to check them to see what went wrong here.